src/deploy/scope_resourcegroup.ts

```diff
@@ -23,7 +23,11 @@ export async function DeployResourceGroupScope(azPath: string, resourceGroupName
             : undefined,
         deploymentMode && deploymentMode != "validate" ? `--mode ${deploymentMode}` : "--mode Incremental",
         deploymentName ? `--name "${deploymentName}"` : undefined,
-        parameters ? `--parameters ${parameters}` : undefined
+        parameters ? `--parameters ${parameters}` : undefined,
+        whatIf ? '--what-if' : undefined,
+        whatIfExcludeChangeTypes ? `--what-if-exclude-change-types ${whatIfExcludeChangeTypes}` : undefined,
+        whatIfResultFormat ? `--what-if-result-format ${whatIfResultFormat}` : undefined,
+        rollbackOnError == true ? '--rollback-on-error' : rollbackOnError == false ? undefined : `--rollback-on-error ${rollbackOnError}`
```

Comment:
does `undefined` work here, or should it be empty string?

Reply:
i would prefer `===` rather than `==`

Reply:
what if the input `rollbackOnError` is an empty string? It is a value which evaluates to `false`, I think we should not roll back in that case
I feel it is good to log in case we are rolling back

Reply:
The rollback will happen automatically so the most we can print is that rollback will happen but that will only be possible after the error took place and rollback has already started.